tests/setup-snapshot.yml

```diff
@@ -0,0 +1,12 @@
+- hosts: all
+  tasks:
+    - name: Set platform/version specific variables
+      include_role:
+        name: linux-system-roles.timesync
+        tasks_from: set_vars.yml
+        public: true
+
+    - name: Install test packages
+      package:
+        name: "{{ timesync_ntp_provider_os_default }}"
```

Comment:
This works, but only by accident, that the name of the provider is the same as the name of the OS package.
Unfortunately, it doesn't look like the timesync role is really suitable for using setup-snapshot - there is no clearly defined list of packages, and there is a *lot* of logic involved to determine the correct packages to install.
I guess what you could do in setup-snapshot is to install `chrony`, and `ntp`, and `linuxptp`, then remove them.  Since not all of these packages will be available on all platforms, just set `no_log: true` and `ignore_errors: true` when installing them (you should not need to ignore errors upon removal).  At least that way the packages will be in the local cache so at least we will get a slight performance improvement.